lib/Frontend/CompilerInvocation.cpp

```diff
@@ -222,26 +222,161 @@ class ArgsToFrontendInputsConverter {
     }
     return false;
   }
+};
+    class FrontendArgsToOptionsConverter {
+    private:
+        DiagnosticEngine &Diags;
+        const llvm::opt::ArgList &Args;
+        FrontendOptions &Opts;
+        
+        llvm::Optional<const std::vector<std::string>>
+        cachedOutputFilenamesFromCommandLineOrFilelist;
+        
+
+
+  // This is a separate function so that it shows up in stack traces.
+  LLVM_ATTRIBUTE_NOINLINE
+  static void debugFailWithAssertion() {
+    // This assertion should always fail, per the user's request, and should
+    // not be converted to llvm_unreachable.
+    assert(0 && "This is an assertion!");
+  }
+
+  // This is a separate function so that it shows up in stack traces.
+  LLVM_ATTRIBUTE_NOINLINE
+  static void debugFailWithCrash() { LLVM_BUILTIN_TRAP; }
+
+  void computeDebugCrashGroup();
```

Comment:
Nothing is being "computed" here, though. `handle` would work.

Reply:
I actually like `handle` for the various option-reading functions too, though that's more of a taste issue instead of being actively misleading. Does it make sense how these are different from `computeModuleName`?

Reply:
It is setting the CrashMode option. These functions all set something, and I think I may have originally called them set<something>. The value that is set is the result of a (possibly trivial) computation, as I tend to frame it. But, I'm happy to name them whatever you like. My preference would be a consistent prefix.